src/Compilers/CSharp/Portable/Symbols/Metadata/PE/ExternalAnnotations.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection.Metadata;
+using System.Text;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE
+{
+    // PROTOTYPE(NullableReferenceTypes): external annotations should be removed or fully designed/productized
+    internal class ExtraAnnotations
+    {
+        // APIs that are useful to annotate:
+        //   1) don't accept null input
+        //   2) can return null
+        //   3) never return null
+        internal static readonly Dictionary<string, ImmutableArray<ImmutableArray<bool>>> Annotations =
+            new Dictionary<string, ImmutableArray<ImmutableArray<bool>>>
+            {
+                { "System.Boolean System.Boolean.Parse(System.String)", Parameters(skip, Nullable(false)) },
+                { "System.Void System.Buffer.BlockCopy(System.Array, System.Int32, System.Array, System.Int32, System.Int32)", Parameters(skip, Nullable(false), skip, Nullable(false), skip, skip) },
+                { "System.Int32 System.Buffer.ByteLength(System.Array)", Parameters(skip, Nullable(false)) },
+                { "System.Byte System.Buffer.GetByte(System.Array, System.Int32)", Parameters(skip, Nullable(false), skip) },
+                { "System.Void System.Buffer.SetByte(System.Array, System.Int32, System.Byte)", Parameters(skip, Nullable(false), skip, skip) },
+                { "System.Byte System.Byte.Parse(System.String)", Parameters(skip, Nullable(false)) },
+                { "System.Byte System.Byte.Parse(System.String, System.IFormatProvider)", Parameters(skip, Nullable(false), skip) },
+                { "System.Byte System.Byte.Parse(System.String, System.Globalization.NumberStyles)", Parameters(skip, Nullable(false), Nullable(false)) },
+                { "System.Byte System.Byte.Parse(System.String, System.Globalization.NumberStyles, System.IFormatProvider)", Parameters(skip, Nullable(false), Nullable(false), skip) },
+                { "System.String System.String.Concat(System.String, System.String)", Parameters(Nullable(false), Nullable(true), Nullable(true)) },
+            };
+
```

Comment:
We will probably want to parse source-like code to produce this so we can have the annotations maintained outside the set of compiler sources.